receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go

```diff
@@ -271,14 +271,17 @@ func (s *processScraper) getProcessMetadata(ctx context.Context) ([]*processMeta
 			continue
 		}
 
-		parentPid, err := parentPid(ctx, handle, pid)
-		if err != nil {
-			errs.AddPartial(0, fmt.Errorf("error reading parent pid for process %q (pid %v): %w", executable.name, pid, err))
+		parentProcessID := int32(0)
+		if !s.config.ExcludeParentPid {
```

Comment:
I think you're right that we don't need a specific config for this. I would like to still make the check conditional, but we can make it conditional on whether the `process.parent_pid` resource attribute is enabled or not.

While on Windows with the new process enumeration method getting ppid no longer costs anything, on other platforms it will still be the old style of doing a certain operation per-process. It's cheap on other platforms, but as a matter of principle I do think any work this scraper does should be conditional on whether the user actually enabled that attribute/metric.